src/transfer/MillikanWhite.cpp

```diff
@@ -42,160 +42,123 @@ using namespace Mutation::Thermodynamics;
-        } else {
-            if (m_thetav < 0.0) {
-                std::cout << "Error: Did not find vibrational temperature for "
-                          << name << "." << std::endl;
-                std::exit(1);
-            }
-
-            // Add Millikan-White data using defaults
-            m_partners.push_back(MillikanWhitePartner(mu, m_thetav));
-        }
+    // Fill in default values
+    double theta_power = std::pow(thetav, 4.0/3.0);
+    size_t offset = (thermo.hasElectrons() ? 1 : 0);
+
+    for (size_t i = 0; i < thermo.nHeavy(); ++i)
+    {
+        double mwi = thermo.speciesMw(i+offset);
+        double mu = 1000.0*m_mw*mwi/(m_mw + mwi); // reduced mass in g/mol
+        m_a[i] = 1.16E-3*std::sqrt(mu)*theta_power;
+        m_b[i] = 0.015*std::pow(mu, 0.25);
     }
 }
 
-//==============================================================================
 
-MillikanWhiteVibrator::MillikanWhiteVibrator(
-    const std::string& name, const class Thermodynamics& thermo)
-    : m_omegav(3.0E-21),
-      m_index(thermo.speciesIndex(name)),
-      m_thetav(loadThetaV(name))
-{
-    // Rely on thetav for all partners so make sure it was found
-    if (m_thetav < 0.0) {
-        std::cout << "Error: Did not find vibrational temperature for "
-                  << name << "." << std::endl;
-        std::exit(1);
-    }
+MillikanWhiteModel::MillikanWhiteModel(const MillikanWhiteModelData& data) :
+    m_data(data) 
+{ }
 
-    const Species& vibrator = thermo.species(name);
-    
-    // Loop over each heavy species in thermo
-    int offset = (thermo.hasElectrons() ? 1 : 0);
-    double mu;
-    
-    for (int i = 0; i < thermo.nHeavy(); ++i) {
-        // Get collision partner
-        const Species& partner = thermo.species(i+offset);
-        
-        // Compute reduced mass of this pair
-        mu = (vibrator.molecularWeight() * partner.molecularWeight()) /
-             (vibrator.molecularWeight() + partner.molecularWeight());
-        
-        // Add Millikan-White data using defaults
-        m_partners.push_back(MillikanWhitePartner(mu, m_thetav));
-    }
+
+double MillikanWhiteModel::relaxationTime(
+    const Mutation::Thermodynamics::Thermodynamics& thermo) const
+{
+    // Millikan-White model for average relaxation time
+    const Eigen::Map<const Eigen::ArrayXd> Xh(
+        thermo.X()+(thermo.hasElectrons() ? 1 : 0), thermo.nHeavy());
+    const double T_fac = std::pow(thermo.T(), -1.0/3.0);
+    const double p_atm = thermo.P() / ONEATM;
+    const double tau_mw = 
+        (Xh*(m_data.a()*(T_fac - m_data.b()) - 18.42).exp()).sum() / 
+        (Xh.sum()*p_atm);
+
+    // Park correction
+    const double ni = thermo.numberDensity() * thermo.X()[m_data.speciesIndex()];
+    const double ci = std::sqrt(8*RU*thermo.T()/(PI*m_data.molecularWeight()));
+    const double tau_park = 1.0/(ni * ci * m_data.limitingCrossSection());
+
+    return tau_mw + tau_park;
```

Comment:
It's a good point, but I think in this specific example, these two terms are always used together (based on the literature sources I've had a look at).  In other words, it is safe to consider the combined model as one, instead of a model with a correction.  If another model is used, it is likely an improvement on the Millikan-White theory altogether (for example the SSH model).  To implement this, it would make sense instead to create a `RelaxationTimeModel` base class with the API of the `MillikanWhiteModel` class and then make `MillikanWhiteModel` and any future models implement this interface.  Then the `Vibrator` class would simply hold a generic `RelaxationTimeModel` pointer.  However, since the Millikan-White model is currently the only model implemented, I think it would be premature to make this abstraction.

Reply:
Looks good to me.